plugins/custom-class/prism-custom-class.js

```diff
@@ -7,24 +7,61 @@ if (
 	return;
 }
 
+/**
+ * @callback ClassMapper
+ * @param {string} className
+ * @param {string} language
+ * @returns {string}
+ */
+/**
+ * @typedef CustomClassOptions
+ * @property {ClassMapper} classMap
+ * @property {string} prefixString
+ */
+
+/** @type {ClassMapper} */
+var defaultClassMap = function (className) { return className; };
+
+/** @type {CustomClassOptions} */
 var options = {
-	classMap: {}
+	classMap: defaultClassMap,
+	prefixString: ''
 };
+
 Prism.plugins.customClass = {
-	map: function map(cm) {
-		options.classMap = cm;
+	/**
+	 * Maps all class names using the given object or map function.
+	 *
+	 * This does not affect the prefix.
+	 *
+	 * @param {Object<string, string> | ClassMapper} classMap
+	 */
+	map: function map(classMap) {
+		if (typeof classMap === 'function') {
+			options.classMap = classMap;
+		} else {
+			options.classMap = function (className) {
+				return classMap[className] || className;
+			};
+		}
 	},
+	/**
+	 * Adds the given prefix to all class names.
+	 *
+	 * @param {string} string
+	 */
 	prefix: function prefix(string) {
 		options.prefixString = string;
 	}
```

Comment:
Thoughts on adding a `reset` method or something so users can revert the `classMap` / `prefix` changes?

Reply:
Sure, sounds good.
But, we should make it so that the user can reset `map` and `prefix` separately. (The user can't read the internal state, so resetting one without the other would be quite a bit harder.)

Maybe a good way is to just provide the initial value via an `initial` property like `Prism.plugins.customClass.map.initial`?

Or maybe something like `Prism.plugins.customClass.map.reset()` so we don't expose any internal state?

Reply:
Sorry, on second thought, I'm not sure it's worth it; I don't see it really being used that often. Let's pass on adding something like that unless we get a request for it.